### PR TITLE
More testing

### PR DESCRIPTION
**Pull request by cavis**
Backfill testing for cms/auth.  Setup some easy way to mock cms/auth for general component tests.

### CODE REVIEW

**Comment by cavis** [comment]
Note: this branch may not actually run without the `package.json` changes from #5.  But I've opted not to push those to this branch, to avoid merge conflicts.

File: app/home/home.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    provide(CmsService, {useClass: MockCmsService})
+  ]);
+
+  const buildComponent = (work: Function): any => {
+    return injectAsync(
+      [TestComponentBuilder, CmsService],
+      (tcb: TestComponentBuilder, cms: MockCmsService) => {
+        cms.mock(['prx:authorization', 'prx:default-account'], {
+          name: 'foobar'
+        });
```
Comment (cavis) [inline app/home/home.component.spec.ts]: This is the actual magic to mock the cms request.  Everything else in this function is angular2 boilerplate for building a component.

**Comment by cavis** [comment]
Phew - several master-merges later, I think this is ready to go again.

A bit of tslint-noise in this one ... but those elses aren't going to cuddle themselves.

File: app/home/home.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import {provide} from 'angular2/core';
+import {it, describe, expect, injectAsync, beforeEachProviders} from 'angular2/testing';
+import {TestComponentBuilder, ComponentFixture, setBaseTestProviders} from 'angular2/testing';
```
Comment (kookster) [inline app/home/home.component.spec.ts]: two imports from the same place, to make the lines shorter?
Is there a shortcut that is just all of them, like `ANGULAR_TESTING_ALL_THE_THINGS`?

Comment (cavis) [inline app/home/home.component.spec.ts]: Hmm... yeah, that's not great.  And I don't see any "all" importers.  Part of my frustration with this component test is the sheer number of things that have to be imported.

You'd prefer a wrapped line then?  Works for me.

Comment (kookster) [inline app/home/home.component.spec.ts]: I'm fine with the aesthetics, maybe we can add our own test file to include that makes this and other imports easier?
Looks like lines 1-7 will be pretty common in many tests.

File: app/shared/cms/haldoc.spec.ts [hunk 3]
```diff
@@ -0,0 +1,185 @@
+import {it, describe, expect, beforeEach} from 'angular2/testing';
+import {Http, Response, ResponseOptions, RequestOptions} from 'angular2/http';
+import {MockBackend} from 'angular2/http/testing';
```
Comment (kookster) [inline app/shared/cms/haldoc.spec.ts]: could be merged with line 1

**Comment by kookster** [comment]
I have only minor comments, pretty much ready to merge this now, and any of them can come in follow up work.

**Pull request merged**